src/blockstore/carv2_disk.rs

```diff
@@ -36,26 +36,25 @@ impl CarV2DiskBlockStore {
             }
 
             // If the file is already a valid CarV2v2
-            if let Ok(mut file) = File::open(path)
-                && let Ok(car) = CarV2::read_bytes(&mut file)
-            {
-                Ok(Self {
-                    path: path.to_path_buf(),
-                    car,
-                })
+            if let Ok(mut file) = File::open(path) {
+                if let Ok(car) = CarV2::read_bytes(&mut file) {
+                    return Ok(Self {
+                        path: path.to_path_buf(),
+                        car,
+                    });
+                }
             }
```

Comment:
calling `new` on the Carv2diskblockstore will check if there is already a valid CAR file at the path given. only if those two `if let`s succeed is that the case. 
otherwise, the `new` method will attempt to create a new file on disk and serialize a new, default CARv2 to it. 
so while i can see the logic here being confusing, the order of operations here still make sense to me

Reply:
Creation should be an intentional choice under known conditions, not a byproduct of a failed read...